### ISSUE TITLE
Infinite while for ConditionalContext with InnerMock Property get access

### ISSUE THREAD

**Issue by sGeeK44** [moq/moq4#292]
Hi,

I would like return several result for a method and result conditionned by inner mock state.
When mocked Method is called in tested class, Conditionnal generate a stackOverflow Exception

To Reproduce behaviour:
In TestFile:

```csharp
var dbResultFound = new Mock<IDataReader>();
dbResultFound.Setup(_ => _.Read()).Returns(true);
dbResultNotFound = new Mock<IDataReader>();
dbResultNotFound.Setup(_ => _.Read()).Returns(false);

var command = new Mock<IDbCommand>();
command.When(() => command.Object.CommandText == "SELECT * FROM TABLE WHERE ID=1")
       .Setup(_ => _.ExecuteReader())
       .Returns(DbResultFound.Object);
command.When(() => command.Object.CommandText == "SELECT * FROM TABLE WHERE ID=2")
       .Setup(_ => _.ExecuteReader())
       .Returns(DbResultNotFound.Object);
```

In TestedFile:

```csharp
command.CommandText = "SELECT * FROM TABLE WHERE ID=2";
command.ExecuteReader();
```

**Comment by kzu** [comment]
Interesting scenario. I'm pretty sure that was not the originally intended design though, so I'm not super surprised it doesn't work ;)

**Comment by stakx** [comment]
The same thing can be achieved as follows:

```csharp
var command = new Mock<IDbCommand>();
command.SetupProperty(cmd => cmd.CommandText);
command.Setup(cmd => cmd.ExecuteReader()).Returns(() =>
    {
        switch (command.Object.CommandText)
        {
            case "SELECT * FROM TABLE WHERE ID=1":
                return dbResultFound.Object;

            case "SELECT * FROM TABLE WHERE ID=2":
            default:
                return dbResultNotFound.Object;
        }
    });
```

I'm not sure if this issue needs to be fixed at all... but it would be nice not to have infinite loops in Moq.

**Comment by stakx** [comment]
This will be fixed in the next release of Moq (version >4.7.63).
